engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/extractor/SqlExtractorErrors.scala

```diff
@@ -0,0 +1,50 @@
+package pl.touk.nussknacker.engine.flink.table.extractor
+
+import pl.touk.nussknacker.engine.flink.table.extractor.SqlStatementReader.SqlStatement
+
+final case class SqlExtractorError(
+    errorType: SqlExtractionErrorType,
+    statement: SqlStatement,
+    exception: Option[Throwable],
+)
+
+object SqlExtractorError {
+
+  def apply(errorType: SqlExtractionErrorType, exception: Option[Throwable] = None)(
+      implicit sqlStatement: SqlStatement
+  ): SqlExtractorError = {
+    SqlExtractorError(errorType, sqlStatement, exception)
+  }
+
+}
+
+sealed trait SqlExtractionErrorType {
+  def message: String
+}
+
+object StatementNotExecuted extends SqlExtractionErrorType {
+
+  // TODO: explain our required syntax - for example we don't handle explicit catalog / db name
+  override val message: String =
+    """
+      |Could not execute sql statement. The statement may be malformed. The statement has to be a CREATE TABLE statement
```

Comment:
Would be nice to have the statement content in this messages :)

Reply:
Same as comment above